server/src/main/scala/za/co/absa/atum/server/api/repository/BaseRepository.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.server.api.repository
+
+import com.github.dwickern.macros.NameOf.nameOf
+import za.co.absa.atum.server.api.exception.DatabaseError
+import za.co.absa.fadb.exceptions.StatusException
+import zio._
+
+trait BaseRepository {
+
+  def handleDbFunctionCall[T, R](
+    dbFuncCall: T => Task[Either[StatusException, R]],
+    inputDTO: T,
+  ): IO[DatabaseError, Either[StatusException, R]] = {
+
+    val operationName = nameOf(dbFuncCall)
```

Comment:
I would rather not use macros where not really needed. Moreover here you are calling the method outside of the zio sequence, that has to be changed. The log statement can be one of the arguments instead and the call to nameOf.

Reply:
yeah I wanted to somehow automate this msgs